packages/sitecore-jss-nextjs/src/utils.ts

```diff
@@ -0,0 +1,60 @@
+import chalk from 'chalk';
+import { isExperienceEditorActive, resetExperienceEditorChromes } from '@sitecore-jss/sitecore-jss';
+
+export const getPublicUrl = (): string => {
+  let url = process.env.PUBLIC_URL;
+  if (url === undefined) {
+    console.warn(
+      `${chalk.yellow.bold(
+        'Warning:'
+      )} An PUBLIC_URL environment variable is not defined. Falling back to http://localhost:3000.`
+    );
+    url = 'http://localhost:3000';
+  } else {
+    try {
+      new URL(url);
+    } catch (error) {
+      throw new Error(`The PUBLIC_URL environment variable '${url}' is not a valid URL.`);
+    }
+  }
+  // Ensure no trailing slash
+  return url.toString().replace(/\/$/, '');
+};
+
+/**
+ * Since Experience Editor does not support Fast Refresh:
+ * 1. Subscribe on events provided by webpack.
+ * 2. Reset experience editor chromes when build is finished
+ * @param {Function} onBuildEnd custom logic to be executed after build is finished
+ */
+export const subscribeOnRebuildAppEvent = (onBuildEnd?: (payload: unknown) => void): void => {
+  if (isExperienceEditorActive()) {
+    const eventSource = new window.EventSource(`${getPublicUrl()}/_next/webpack-hmr`);
+
+    window.addEventListener('beforeunload', () => eventSource.close());
+
+    eventSource.onopen = () => console.log('[Experience Editor HMR Listener] Online');
```

Comment:
Should maybe use Next.js's term "[Fast Refresh](https://nextjs.org/docs/basic-features/fast-refresh)" instead of "HMR" in our logs.

Reply:
Yes, it's more related to Fast Refresh than to Hot module replacement term